### ISSUE TITLE
XCode produces binaries that aren't able to handle blanks in quoted parameters

### ISSUE THREAD

**Issue by leowmjw** [arangodb/arangodb#1779]
It seems like arangoimp is not able to take in space in the file parameter even though it is quoted.

ArangoDB 2.8.3; on OSX; started using the ArangoDB UI

Example:
```
leow$ arangoimp --collection=example --server.database=myapp --file="./DR-30112015/Waktu Mesyuarat dan Urusan Dibebaskan Daripada Peraturan Mesyuarat/speakers.json"
```

Output:
```
Connected to ArangoDB 'tcp://127.0.0.1:8000', version 2.8.3, database: 'myapp', username: 'root'
----------------------------------------
database:         myapp
collection:       example
create:           no
file:             ./DR-30112015/Waktu
type:             json
connect timeout:  5
request timeout:  1200
----------------------------------------
Cannot open file './DR-30112015/Waktu'. File not found.
```

**Comment by jsteemann** [comment]
We tried it with the same filename and it worked fine on Linux and MacOS X here.
As the filename is truncated to `./DR-30112015/Waktu` in your case, it seems like the file name argument contains a null byte (ASCII NUL, character code 0) somewhere. This would explain truncation. NUL bytes are normally not printed, so it's not apparent if the filename argument contains one or not.

Can you try with another filename/directory name that contains spaces but no NUL bytes, e.g.
```bash
mkdir -p "test directory/test sub directory"
echo "{}" > "test directory/test sub directory/test file.json"
arangoimp --file "test directory/test sub directory/test file.json" --collection test --create-collection true
```
If this works, then this would support the theory that somehow the filename argument in your case contained some non-printable characters.

**Comment by leowmjw** [comment]
@jsteemann Unfortunately did not seem to work; I manually copied over a JSON file to a file with filename containing space but seem to have same problem.  I am still on OSX Mavericks if that matters.

Create file:
```
leow$ cp -a /tmp/bob.json "/tmp/bob is bool.json"

leow$ ls /tmp/bob*
/tmp/bob is bool.json	/tmp/bob.json   
        
```

When run:
```
leow$ arangoimp --collection=transcript --create-collection=true --server.database=myapp --file="/tmp/bob is bool.json"
Connected to ArangoDB 'tcp://127.0.0.1:8000', version 2.8.3, database: 'myapp', username: 'root'
----------------------------------------
database:         myapp
collection:       transcript
create:           yes
file:             /tmp/bob
type:             json
connect timeout:  5
request timeout:  1200
----------------------------------------
Cannot open file '/tmp/bob'. File not found.
```

**Comment by baslr** [comment]
@leowmjw @jsteemann can you check your IFS? `echo "$IFS" | cat -vte`

**Comment by jsteemann** [comment]
```
echo "$IFS" | cat -vte
```
```
 ^I$
$
```

**Comment by leowmjw** [comment]
My IFS:
```
leow$ echo "$IFS" | cat -vte
 ^I$
$
```

**Comment by baslr** [comment]
I tested it by myself:

devel@bb8f603d7f1c89ba10839234c940788f4c78f3fa:
```
./arangoimp --collection example --file="abc def"
Please specify a password:
Connected to ArangoDB 'tcp://127.0.0.1:8529', version 2.8.4, database: '_system', username: 'root'
----------------------------------------
database:         _system
collection:       example
create:           no
file:             abc def
type:             json
connect timeout:  5
request timeout:  1200
----------------------------------------
2016-03-13T05:03:51Z [94755] FATAL Cannot open file 'abc def'. File not found.
```

2.8.4
```
/Applications/ArangoDB.app/Contents/Resources/bin/arangoimp --collection example --file="abc def"
Connected to ArangoDB 'tcp://127.0.0.1:8529', version 2.8.4, database: '_system', username: 'root'
----------------------------------------
database:         _system
collection:       example
create:           no
file:             abc
type:             json
connect timeout:  5
request timeout:  1200
----------------------------------------
Cannot open file 'abc'. File not found.
```

**Comment by baslr** [comment]
more interesting, local compiled v2.8.4:
```
./arangoimp --collection example --file="abc def"
Please specify a password:
Connected to ArangoDB 'tcp://127.0.0.1:8529', version 2.8.4, database: '_system', username: 'root'
----------------------------------------
database:         _system
collection:       example
create:           no
file:             abc def
type:             json
connect timeout:  5
request timeout:  1200
----------------------------------------
Cannot open file 'abc def'. File not found.
```

**Comment by baslr** [comment]
I checked the OS X Apps:
```
CLI-App:
/Volumes/ArangoDB-CLI-2.8.5-Darwin/ArangoDB-CLI.app/Contents/MacOS/opt/arangodb/bin/arangoimp --collection example --file="abc def"
Please specify a password:
Connected to ArangoDB 'tcp://127.0.0.1:8529', version 2.8.4, database: '_system', username: 'root'
----------------------------------------
database:         _system
collection:       example
create:           no
file:             abc def
type:             json
connect timeout:  5
request timeout:  1200
----------------------------------------
Cannot open file 'abc def'. File not found.

# GUI-App
/Volumes/ArangoDB/ArangoDB.app/Contents/Resources/bin/arangoimp --collection example --file="abc def"
Connected to ArangoDB 'tcp://127.0.0.1:8529', version 2.8.4, database: '_system', username: 'root'
----------------------------------------
database:         _system
collection:       example
create:           no
file:             abc
type:             json
connect timeout:  5
request timeout:  1200
----------------------------------------
Cannot open file 'abc'. File not found.
```

Conclusion: the OS X GUI-App cant handle whitespace in fs paths.

**Comment by jsteemann** [comment]
@baslr : interesting. I think I wouldn't have figured it out myself.

**Comment by dothebart** [comment]
since this seems to be an issue with the infrastucture of MacOS X, is there anything else we can do about this than documenting it? Else I'd like to close this now.

**Comment by baslr** [comment]
Are there any differences in the cli and gui app (source / compiling)?

**Comment by leowmjw** [comment]
@dothebart Yeah, I think the issue should be kept opened until both CLI and GUI are compiled the same way. You don't want users who develop using the GUI facing unexpected problems when it comes to production deployment. For an officially supported platform, that would be the least expected, this should be labeled a bug under OSX, probably should test for Windows too.

**Comment by dothebart** [comment]
@baslr what do you reference as GUI and which as CLI app? Are both built via homebrew?

**Comment by leowmjw** [comment]
@dothebart Just for completeness; the ArangoDB CLI + App I use are the official DMG from ArangoDB site.

**Comment by baslr** [comment]
@dothebart I checked the apps from https://www.arangodb.com/download/

**Comment by dothebart** [comment]
Ok, App uses clang via cmake/xcodebuild while CLI uses clang via cmake/make

**Comment by jsteemann** [comment]
It's not an issue with command-line parsing.
The issue is caused by the parameters to the executables in the Mac app version being wrapped in an extra shell script, which passes parameters with `$*` instead of `$@`.
This will be fixed in the next 2.8 release.

**Comment by dothebart** [comment]
fixed with the availability of 2.8.9 since 2016-05-13.
